lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -6766,19 +6814,51 @@ void EmitDestructuredObject(ParseNode *lhs,
 
     if (pnode1 != nullptr)
     {
-        Assert(pnode1->nop == knopList || pnode1->nop == knopObjectPatternMember);
+        Js::RegSlot propIdArrLocation = 0;
+        Js::RegSlot computedPropIdArrLocation = 0;
+        if (hasRest)
+        {
+            int extraAlloc = 0, auxSize = 0;
+            if (Int32Math::Mul(staticCount, sizeof(Js::PropertyId), &extraAlloc)
```

Comment:
Should this be `Int32Math` or `Uint32Math`? Although I hope there aren't that many objects with over a billion properties that would make this relevant...

Reply:
Probably Uint...good catch!

---
In reply to: [204227361](https://github.com/Microsoft/ChakraCore/pull/5498#discussion_r204227361) [](ancestors = 204227361)